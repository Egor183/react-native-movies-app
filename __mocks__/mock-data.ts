import MockAdapter from 'axios-mock-adapter';
import { api as axios } from '@src/services/api.services';

export const mock = new MockAdapter(axios);

export const MOCK_MOVIES_LIST = [
  {
    backdrop:
      'https://wookie.codesubmit.io/static/backdrops/d6822b7b-48bb-4b78-ad5e-9ba04c517ec8.jpg',
    cast: ['Christian Bale', 'Heath Ledger', 'Aaron Eckhart'],
    classification: '13+',
    director: 'Christopher Nolan',
    genres: ['Action', 'Crime', 'Drama'],
    id: 'd6822b7b-48bb-4b78-ad5e-9ba04c517ec8',
    imdb_rating: 9.0,
    length: '2h 32min',
    overview:
      'Batman raises the stakes in his war on crime. With the help of Lt. Jim Gordon and District Attorney Harvey Dent, Batman sets out to dismantle the remaining criminal organizations that plague the streets. The partnership proves to be effective, but they soon find themselves prey to a reign of chaos unleashed by a rising criminal mastermind known to the terrified citizens of Gotham as the Joker.',
    poster:
      'https://wookie.codesubmit.io/static/posters/d6822b7b-48bb-4b78-ad5e-9ba04c517ec8.jpg',
    released_on: '2008-07-16T00:00:00',
    slug: 'the-dark-knight-2008',
    title: 'The Dark Knight',
  },
  {
    backdrop:
      'https://wookie.codesubmit.io/static/backdrops/0bf56a90-5f7b-4108-9d0b-01ad77f0a310.jpg',
    cast: ['Malcolm McDowell', 'Patick Magee', 'Michael Bates'],
    classification: '18+',
    director: 'Stanley Kubrick',
    genres: ['Crime', 'Drama', 'Sci-Fi'],
    id: '0bf56a90-5f7b-4108-9d0b-01ad77f0a310',
    imdb_rating: 8.3,
    length: '2h 16min',
    overview:
      'Demonic gang-leader Alex goes on the spree of rape, mugging and murder with his pack of "droogs". But he\'s a boy who also likes Beethoven\'s Ninth and a bit of "the old in-out, in-out". He later finds himself at the mercy of the state and its brainwashing experiment designed to take violence off the streets.',
    poster:
      'https://wookie.codesubmit.io/static/posters/0bf56a90-5f7b-4108-9d0b-01ad77f0a310.jpg',
    released_on: '1971-12-18T00:00:00',
    slug: 'a-clockwork-orange-1971',
    title: 'A Clockwork Orange',
  },
];
