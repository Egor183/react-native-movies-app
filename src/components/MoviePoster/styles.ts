import { ScaledSheet } from 'react-native-size-matters';
import { setModerateScale } from '@src/services/scale.services';
import { COLORS } from '@src/constants/colors.constants';

export default ScaledSheet.create({
  movieContainer: {
    backgroundColor: COLORS.GRAY_COLOR,
    alignItems: 'center',
    justifyContent: 'center',
  },

  noImageDescription: {
    fontWeight: '600',
    padding: setModerateScale(8),
  },

  image: {
    height: '100%',
    width: '100%',
  },
});
