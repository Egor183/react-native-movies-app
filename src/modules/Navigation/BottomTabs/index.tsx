import React, { memo } from 'react';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { NAVIGATORS } from '@src/constants/routes.constants';
import { useTabBar } from './hooks/useTabBar.hook';
import MoviesStackNavigator from '../Movies';
import Search from '../Search';

const Tab = createBottomTabNavigator();

const BottomTabNavigator = () => {
  const tabBar = useTabBar();

  return (
    <Tab.Navigator
      tabBar={tabBar}
      initialRouteName={NAVIGATORS.MOVIES_NAVIGATOR}
      screenOptions={{
        headerShown: false,
        tabBarHideOnKeyboard: true,
      }}>
      <Tab.Screen
        name={NAVIGATORS.MOVIES_NAVIGATOR}
        component={MoviesStackNavigator}
      />
      <Tab.Screen name={NAVIGATORS.SEARCH_NAVIGATOR} component={Search} />
    </Tab.Navigator>
  );
};

export default memo(BottomTabNavigator);
