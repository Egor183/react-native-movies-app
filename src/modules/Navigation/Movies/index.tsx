import React, { memo } from 'react';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { MOVIES_STACK_ROUTES } from '@src/constants/routes.constants';
import Movies from '@src/screens/Movies';
import MovieDetails from '@src/screens/MovieDetails';

const Stack = createNativeStackNavigator();

const MoviesStackNavigator = () => (
  <Stack.Navigator
    initialRouteName={MOVIES_STACK_ROUTES.MOVIES}
    screenOptions={{ headerShown: false }}>
    <Stack.Screen name={MOVIES_STACK_ROUTES.MOVIES} component={Movies} />
    <Stack.Screen
      name={MOVIES_STACK_ROUTES.MOVIE_DETAILS}
      component={MovieDetails}
    />
  </Stack.Navigator>
);

export default memo(MoviesStackNavigator);
