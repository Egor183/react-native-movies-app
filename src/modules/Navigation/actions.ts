import { createNavigationContainerRef } from '@react-navigation/native';

export const navigationRef = createNavigationContainerRef();

export const navigate = (name: string, params?: any): void => {
  navigationRef.current?.navigate(name as never, params as never);
};

export const goBack = () => {
  const ref = navigationRef.current;

  if (ref?.canGoBack()) {
    ref.goBack();
  }
};
