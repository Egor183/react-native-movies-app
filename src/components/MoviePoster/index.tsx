import React, { memo } from 'react';
import { Image, Pressable, Text } from 'react-native';
import { moderateScale } from 'react-native-size-matters';
import { DEFAULT_POSTER_WIDTH } from './constants/cnstants';
import { calculateFontSize, calculatePosterSizes } from './helpers/helpers';

import styles from './styles';

type Props = {
  onPress?: () => void | null;
  uri?: string;
  gap?: number;
  width?: number;
};

const MoviePoster: React.FC<Props> = ({
  onPress = null,
  uri,
  gap = 0,
  width = DEFAULT_POSTER_WIDTH,
}) => (
  <Pressable
    style={[
      styles.movieContainer,
      {
        marginLeft: moderateScale(gap),
        ...calculatePosterSizes(width),
      },
    ]}
    onPress={onPress ? onPress : null}>
    {uri ? (
      <Image
        style={styles.image}
        source={{
          uri,
        }}
      />
    ) : (
      <Text
        style={[
          styles.noImageDescription,
          { fontSize: calculateFontSize(width) },
        ]}>
        No poster...
      </Text>
    )}
  </Pressable>
);

export default memo(MoviePoster);
