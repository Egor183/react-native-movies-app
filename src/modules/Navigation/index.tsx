import React, { memo } from 'react';
import { NavigationContainer } from '@react-navigation/native';
import {
  initialWindowMetrics,
  SafeAreaProvider,
} from 'react-native-safe-area-context';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import ErrorModal from '@src/components/ErrorModal';
import { MODAL_ROUTES, NAVIGATORS } from '@src/constants/routes.constants';
import BottomTabNavigator from './BottomTabs';
import { navigationRef } from './actions';

const RootStack = createNativeStackNavigator();

const Navigator = () => (
  <SafeAreaProvider initialMetrics={initialWindowMetrics}>
    <NavigationContainer ref={navigationRef}>
      <RootStack.Navigator screenOptions={{ headerShown: false }}>
        <RootStack.Group>
          <RootStack.Screen
            name={NAVIGATORS.BOTTOM_TAB_NAVIGATOR}
            component={BottomTabNavigator}
          />
        </RootStack.Group>
        <RootStack.Group
          screenOptions={{
            presentation: 'modal',
            gestureEnabled: true,
          }}>
          <RootStack.Screen name={MODAL_ROUTES.ERROR} component={ErrorModal} />
        </RootStack.Group>
      </RootStack.Navigator>
    </NavigationContainer>
  </SafeAreaProvider>
);

export default memo(Navigator);
