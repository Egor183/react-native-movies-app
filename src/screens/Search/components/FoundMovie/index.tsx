import React, { memo } from 'react';
import { Pressable, Text, View } from 'react-native';
import MoviePoster from '@src/components/MoviePoster';

import styles from './styles';

type Props = {
  uri: string;
  title: string;
  classification: string;
  rating: number;
  description: string;
  onPress: () => void;
};

const FoundMovie: React.FC<Props> = ({
  uri,
  title,
  classification,
  rating,
  description,
  onPress,
}) => (
  <Pressable style={styles.movieContainer} onPress={onPress}>
    <MoviePoster width={50} uri={uri} />
    <View style={styles.movieInformationContainer}>
      <Text style={styles.title}>{title}</Text>
      <Text style={styles.baseInfo}>Classification: {classification}</Text>
      <Text style={styles.baseInfo}>Rating: {rating}</Text>
      <Text numberOfLines={2} style={styles.baseInfo}>
        Description: {description}
      </Text>
    </View>
  </Pressable>
);

export default memo(FoundMovie);
