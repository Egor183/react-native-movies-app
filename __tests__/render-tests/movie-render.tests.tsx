import React from 'react';
import {
  MOVIES_STACK_ROUTES,
  NAVIGATORS,
} from '@src/constants/routes.constants';
import { navigate } from '@src/modules/Navigation/actions';
import { FlatList } from 'react-native';
import renderer from 'react-test-renderer';
import { MovieType, PosterType } from '@src/types/movies.types';
import FoundMovie from '@src/screens/Search/components/FoundMovie';
import MoviePoster from '@src/components/MoviePoster';
import { MOCK_MOVIES_LIST } from '../../__mocks__/mock-data';

describe('Movies renders test', () => {
  it('Should render found movies list', () => {
    type RenderItemProps = {
      item: MovieType;
    };

    const renderItem = ({ item }: RenderItemProps) => {
      const handlePress = () => {
        navigate(NAVIGATORS.MOVIES_NAVIGATOR, {
          screen: MOVIES_STACK_ROUTES.MOVIE_DETAILS,
          params: { id: item.id },
        });
      };

      return (
        <FoundMovie
          uri={item.poster}
          title={item.title}
          classification={item.classification}
          rating={item.imdb_rating}
          description={item.overview}
          onPress={handlePress}
        />
      );
    };

    const foundMovies = renderer
      .create(
        <FlatList
          renderItem={renderItem}
          keyExtractor={item => item.id}
          data={MOCK_MOVIES_LIST}
          showsVerticalScrollIndicator={false}
        />,
      )
      .toJSON();

    expect(foundMovies).toMatchSnapshot();
  });

  it('Should render posters list', () => {
    type RenderItemProps = { item: PosterType };

    const renderItem = ({ item }: RenderItemProps) => {
      const handlePress = () => {
        navigate(MOVIES_STACK_ROUTES.MOVIE_DETAILS, { id: item.id });
      };

      return <MoviePoster onPress={handlePress} uri={item.poster} gap={10} />;
    };

    const foundMovies = renderer
      .create(
        <FlatList
          renderItem={renderItem}
          keyExtractor={item => item.id}
          data={MOCK_MOVIES_LIST}
          horizontal
          showsHorizontalScrollIndicator={false}
          initialNumToRender={5}
          maxToRenderPerBatch={5}
        />,
      )
      .toJSON();

    expect(foundMovies).toMatchSnapshot();
  });
});
