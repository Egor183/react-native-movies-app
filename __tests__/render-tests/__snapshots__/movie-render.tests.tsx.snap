// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies renders test Should render found movies list 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "backdrop": "https://wookie.codesubmit.io/static/backdrops/d6822b7b-48bb-4b78-ad5e-9ba04c517ec8.jpg",
        "cast": Array [
          "Christian Bale",
          "Heath Ledger",
          "Aaron Eckhart",
        ],
        "classification": "13+",
        "director": "Christopher Nolan",
        "genres": Array [
          "Action",
          "Crime",
          "Drama",
        ],
        "id": "d6822b7b-48bb-4b78-ad5e-9ba04c517ec8",
        "imdb_rating": 9,
        "length": "2h 32min",
        "overview": "Batman raises the stakes in his war on crime. With the help of Lt. Jim Gordon and District Attorney Harvey Dent, Batman sets out to dismantle the remaining criminal organizations that plague the streets. The partnership proves to be effective, but they soon find themselves prey to a reign of chaos unleashed by a rising criminal mastermind known to the terrified citizens of Gotham as the Joker.",
        "poster": "https://wookie.codesubmit.io/static/posters/d6822b7b-48bb-4b78-ad5e-9ba04c517ec8.jpg",
        "released_on": "2008-07-16T00:00:00",
        "slug": "the-dark-knight-2008",
        "title": "The Dark Knight",
      },
      Object {
        "backdrop": "https://wookie.codesubmit.io/static/backdrops/0bf56a90-5f7b-4108-9d0b-01ad77f0a310.jpg",
        "cast": Array [
          "Malcolm McDowell",
          "Patick Magee",
          "Michael Bates",
        ],
        "classification": "18+",
        "director": "Stanley Kubrick",
        "genres": Array [
          "Crime",
          "Drama",
          "Sci-Fi",
        ],
        "id": "0bf56a90-5f7b-4108-9d0b-01ad77f0a310",
        "imdb_rating": 8.3,
        "length": "2h 16min",
        "overview": "Demonic gang-leader Alex goes on the spree of rape, mugging and murder with his pack of \\"droogs\\". But he's a boy who also likes Beethoven's Ninth and a bit of \\"the old in-out, in-out\\". He later finds himself at the mercy of the state and its brainwashing experiment designed to take violence off the streets.",
        "poster": "https://wookie.codesubmit.io/static/posters/0bf56a90-5f7b-4108-9d0b-01ad77f0a310.jpg",
        "released_on": "1971-12-18T00:00:00",
        "slug": "a-clockwork-orange-1971",
        "title": "A Clockwork Orange",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  showsVerticalScrollIndicator={false}
  stickyHeaderIndices={Array []}
  viewabilityConfigCallbackPairs={Array []}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexDirection": "row",
            "marginTop": 22.213235294117645,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#a49e9e",
                "justifyContent": "center",
              },
              Object {
                "height": 157.14285714285714,
                "marginLeft": 0,
                "width": 78.57142857142857,
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "uri": "https://wookie.codesubmit.io/static/posters/d6822b7b-48bb-4b78-ad5e-9ba04c517ec8.jpg",
              }
            }
            style={
              Object {
                "height": "100%",
                "width": "100%",
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "justifyContent": "space-between",
              "marginLeft": 15.714285714285714,
            }
          }
        >
          <Text
            style={
              Object {
                "fontFamily": "Jost-SemiBold",
                "fontSize": 25.142857142857142,
              }
            }
          >
            The Dark Knight
          </Text>
          <Text
            style={
              Object {
                "fontFamily": "Jost-Regular",
                "fontSize": 22,
                "width": 345.71428571428567,
              }
            }
          >
            Classification: 
            13+
          </Text>
          <Text
            style={
              Object {
                "fontFamily": "Jost-Regular",
                "fontSize": 22,
                "width": 345.71428571428567,
              }
            }
          >
            Rating: 
            9
          </Text>
          <Text
            numberOfLines={2}
            style={
              Object {
                "fontFamily": "Jost-Regular",
                "fontSize": 22,
                "width": 345.71428571428567,
              }
            }
          >
            Description: 
            Batman raises the stakes in his war on crime. With the help of Lt. Jim Gordon and District Attorney Harvey Dent, Batman sets out to dismantle the remaining criminal organizations that plague the streets. The partnership proves to be effective, but they soon find themselves prey to a reign of chaos unleashed by a rising criminal mastermind known to the terrified citizens of Gotham as the Joker.
          </Text>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexDirection": "row",
            "marginTop": 22.213235294117645,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#a49e9e",
                "justifyContent": "center",
              },
              Object {
                "height": 157.14285714285714,
                "marginLeft": 0,
                "width": 78.57142857142857,
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "uri": "https://wookie.codesubmit.io/static/posters/0bf56a90-5f7b-4108-9d0b-01ad77f0a310.jpg",
              }
            }
            style={
              Object {
                "height": "100%",
                "width": "100%",
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "justifyContent": "space-between",
              "marginLeft": 15.714285714285714,
            }
          }
        >
          <Text
            style={
              Object {
                "fontFamily": "Jost-SemiBold",
                "fontSize": 25.142857142857142,
              }
            }
          >
            A Clockwork Orange
          </Text>
          <Text
            style={
              Object {
                "fontFamily": "Jost-Regular",
                "fontSize": 22,
                "width": 345.71428571428567,
              }
            }
          >
            Classification: 
            18+
          </Text>
          <Text
            style={
              Object {
                "fontFamily": "Jost-Regular",
                "fontSize": 22,
                "width": 345.71428571428567,
              }
            }
          >
            Rating: 
            8.3
          </Text>
          <Text
            numberOfLines={2}
            style={
              Object {
                "fontFamily": "Jost-Regular",
                "fontSize": 22,
                "width": 345.71428571428567,
              }
            }
          >
            Description: 
            Demonic gang-leader Alex goes on the spree of rape, mugging and murder with his pack of "droogs". But he's a boy who also likes Beethoven's Ninth and a bit of "the old in-out, in-out". He later finds himself at the mercy of the state and its brainwashing experiment designed to take violence off the streets.
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`Movies renders test Should render posters list 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "backdrop": "https://wookie.codesubmit.io/static/backdrops/d6822b7b-48bb-4b78-ad5e-9ba04c517ec8.jpg",
        "cast": Array [
          "Christian Bale",
          "Heath Ledger",
          "Aaron Eckhart",
        ],
        "classification": "13+",
        "director": "Christopher Nolan",
        "genres": Array [
          "Action",
          "Crime",
          "Drama",
        ],
        "id": "d6822b7b-48bb-4b78-ad5e-9ba04c517ec8",
        "imdb_rating": 9,
        "length": "2h 32min",
        "overview": "Batman raises the stakes in his war on crime. With the help of Lt. Jim Gordon and District Attorney Harvey Dent, Batman sets out to dismantle the remaining criminal organizations that plague the streets. The partnership proves to be effective, but they soon find themselves prey to a reign of chaos unleashed by a rising criminal mastermind known to the terrified citizens of Gotham as the Joker.",
        "poster": "https://wookie.codesubmit.io/static/posters/d6822b7b-48bb-4b78-ad5e-9ba04c517ec8.jpg",
        "released_on": "2008-07-16T00:00:00",
        "slug": "the-dark-knight-2008",
        "title": "The Dark Knight",
      },
      Object {
        "backdrop": "https://wookie.codesubmit.io/static/backdrops/0bf56a90-5f7b-4108-9d0b-01ad77f0a310.jpg",
        "cast": Array [
          "Malcolm McDowell",
          "Patick Magee",
          "Michael Bates",
        ],
        "classification": "18+",
        "director": "Stanley Kubrick",
        "genres": Array [
          "Crime",
          "Drama",
          "Sci-Fi",
        ],
        "id": "0bf56a90-5f7b-4108-9d0b-01ad77f0a310",
        "imdb_rating": 8.3,
        "length": "2h 16min",
        "overview": "Demonic gang-leader Alex goes on the spree of rape, mugging and murder with his pack of \\"droogs\\". But he's a boy who also likes Beethoven's Ninth and a bit of \\"the old in-out, in-out\\". He later finds himself at the mercy of the state and its brainwashing experiment designed to take violence off the streets.",
        "poster": "https://wookie.codesubmit.io/static/posters/0bf56a90-5f7b-4108-9d0b-01ad77f0a310.jpg",
        "released_on": "1971-12-18T00:00:00",
        "slug": "a-clockwork-orange-1971",
        "title": "A Clockwork Orange",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={true}
  initialNumToRender={5}
  keyExtractor={[Function]}
  maxToRenderPerBatch={5}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  showsHorizontalScrollIndicator={false}
  stickyHeaderIndices={Array []}
  viewabilityConfigCallbackPairs={Array []}
>
  <View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#a49e9e",
              "justifyContent": "center",
            },
            Object {
              "height": 282.8571428571429,
              "marginLeft": 15.714285714285714,
              "width": 141.42857142857144,
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "uri": "https://wookie.codesubmit.io/static/posters/d6822b7b-48bb-4b78-ad5e-9ba04c517ec8.jpg",
            }
          }
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        />
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#a49e9e",
              "justifyContent": "center",
            },
            Object {
              "height": 282.8571428571429,
              "marginLeft": 15.714285714285714,
              "width": 141.42857142857144,
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "uri": "https://wookie.codesubmit.io/static/posters/0bf56a90-5f7b-4108-9d0b-01ad77f0a310.jpg",
            }
          }
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        />
      </View>
    </View>
  </View>
</RCTScrollView>
`;
